### ISSUE TITLE
SpectrumProcessor saving of spectra should maybe be one by one

### ISSUE THREAD

**Issue by florian-huber** [matchms/matchms#646]
Currently the `SpectrumProcessor`  will first process all spectra (`self.process_spectrum`) and only at the end write them to file (`save_spectra(processed_spectrums, cleaned_spectra_file)`).

For larger workflows this seems not ideal because it can crash at a very late stage (just happened to me...).
I think it would be better to write each spectrum to the file once processed, at least for formats where this is possible such as `.msp` and `.mgf`.

**Comment by hechth** [comment]
I agree - this is also already possible with the streaming write we implemented for those formats - we should add the json implementation though at some point.

**Comment by acquayefrank** [comment]
It seems that this issue has been resolved by this PR: https://github.com/matchms/matchms/pull/651. Hence, I think it should be closed.
